File: 455.assign-cookies.js
/*
 * @lc app=leetcode id=455 lang=javascript
 *
 * [455] Assign Cookies
 */

// @lc code=start
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function(g, s) {
    let count = {}
    let res = 0
    s.forEach(cookie => {
        count[cookie] ? count[cookie]++ : count[cookie] = 1
    })
    g.forEach(cookie => {
        if (count[cookie] - 1 >= 0) {
            res++
            count[cookie]--
        }
    })
    return res
};
// @lc code=end

File: 455.assign-cookies.py
#
# @lc app=leetcode id=455 lang=python
#
# [455] Assign Cookies
#

# @lc code=start
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """

        # Approach 1: Greedy
        # Time  complexity: O(nlogn)
        # Space complexity: O(1)
        g.sort()
        s.sort()
        res = 0
        i = j = 0
        while i < len(g) and j < len(s):
            if g[i] <= s[j]:
                res += 1
                i += 1
            j += 1
        return res

# @lc code=end
